extensions/amp-date-picker/amp-date-picker.md

```diff
@@ -0,0 +1,307 @@
+with an existing element in the form, an error is emitted.
+
+##### end-input-selector (optional)
+
+A query selector for a date range picker's input. If this is omitted,
+the date picker automatically generates a hidden input field. It will assign it
+a name `end-date` or `${id}-end-date` using the date picker's id. If either of these conflict
+with an existing element in the form, an error is emitted.
+
+##### min (optional)
+
+The earliest date that the user may select. The default value is the current date.
+
+##### max (optional)
+
+The latest date that the user may select. The default value is no end date.
+
+#####  month-format (optional)
+
+The format to use for displaying the month in the calendar view. The default format is: `"MMMM YYYY"`.
+
+##### format (optional)
+
+The format to use for displaying and parsing the date in the input boxes. Default: `"YYYY-MM-DD"`.
+
+##### week-day-format (optional)
+
+The format to use for displaying the day of the week in the calendar view. The default format is the non-ISO-standard single character weekday.
+
+##### locale (optional)
+
+The locale to use for rendering the calendar view. The default locale is `"en"`.
+
+##### number-of-months (optional)
+
+The number of months to display at one time in the calendar view. The default is `"1"`.
+
+##### first-day-of-week (optional)
+
+The day to specify as the first day of the week (0-6). The default value is `"0"` (Sunday).
+
+##### blocked (optional)
+
+A list of ISO 8601 dates or RFC 5545 RRULE repeating dates to prevent the user from selecting on the calendar.
+
+##### highlighted (optional)
+
+A list of ISO 8601 dates or RFC 5545 RRULE repeating dates to specially style as highlighted to draw the user's attention. Default styling is a blue dot on the date.
+
+##### day-size (optional)
+
+The size in `px` of the date cells in the calendar view table. The default is `39`.
+
+##### allow-blocked-ranges (optional)
+
+If present, this attribute prevents the user from selecting a range with a blocked date. By default, this attribute is not present.
+
+##### src (optional)
+
+If present, `amp-date-picker` makes a request for JSON data to populate the `highlighted` and `blocked` lists, as well as matching templates in the document to lists of dates.
+
+```json
+{
+  "blocked": ["2018-02-14"],
+  "highlighted": ["2018-02-15"],
+  "templates": [
+    {
+      "selector": "#my-template-id",
+      "dates": ["2018-01-01"]
+    },
+    {
+      "selector": "#my-second-template-id",
+      "dates": [
+        "2018-01-01",
+        "FREQ=WEEKLY;DTSTART=20180201T150000Z;COUNT=30;WKST=SU;BYDAY=TU"
+      ]
+    },
+    {
+      "selector": "#my-default-template-id"
+    }
+  ]
+}
+```
+
+```html
+<amp-date-picker src="https://www.example.com/date-data.json">
+  <template id="my-template-id">⚡️</template>
+  <template id="my-second-template-id">🌮</template>
+  <template id="my-default-template-id">{{D}}</template>
+</amp-date-picker>
```

Comment:
I need to add info about the template API, e.g. `info-template` `date-template dates=""`